src/lightning/pytorch/overrides/distributed.py

```diff
@@ -245,10 +247,10 @@ def __iter__(self) -> Iterator:
         return (self.dataset[index] for index in super().__iter__())
 
 
-class _IndexBatchSamplerWrapper(BatchSampler):
```

Comment:
The typing changes here are unrelated and could go into a separate PR if you preferred that.